conans/client/rest/file_downloader.py

```diff
@@ -86,24 +95,43 @@ def write_chunks(chunks, path):
                 ret = bytes(ret_data)
             return ret, downloaded_size
 
+        def get_total_length():
+            content_range = response.headers.get("Content-Range", "")
+            match = re.match(r"^bytes (\d+)-(\d+)/(\d+)", content_range)
+            if match:
+                assert range_start == int(match.group(1))
+                return int(match.group(3))
+            else:
+                assert range_start == 0
```

Comment:
This assert would be unnecessary if we checked for the Content-Range header only if range_start != 0.